src/mem.c

```diff
@@ -97,9 +97,10 @@ void tell_mem_status(char *nick)
 #ifdef DEBUG_MEM
   float per;
 
-  per = ((lastused * 1.0) / (memtbl_size * 1.0)) * 100.0;
-  dprintf(DP_HELP, "NOTICE %s :Memory table usage: %d/%d (%.1f%% full)\n",
-          nick, lastused, memtbl_size, per);
+  per = 100.0 * (lastused * 1.0) / (MEMTBLSIZE_MAX * 1.0);
```

Comment:
the reason 100.0 was at the end was overflow-prevention (100 * lastused can overflow while lastused/MEMTBLSIZE_MAX * 100 might not)
`per = 1.0 * lastused / MEMTBLSIZE_MAX * 100` would work, too, or cast one to (double), or revert to what it was :-)

Reply:
isn't the result kinda wrong in both cases if something maxes out? i was optimizing for rounding errors, thats why i always multiply before division. but i see your point. the error margin is bigger for overflow. so i will change the order again...